File: Codigos estudiantes por lenguaje/PY/Bryann Valderrama/BitWise/palindromosBinarios.py
from sys import stdin, stdout, maxsize
rl = stdin.readline
wr = stdout.write


def iskthBitSet(x, k):
    return 1 if (x & (1 << (k-1))) > 0 else 0


def leftMostSetBit(x):
    cont = 0
    while x > 0:
        cont += 1
        x = x >> 1
    return cont


def isBinPal(x):
    l = leftMostSetBit(x)
    r = 1
    while l > r:
        if iskthBitSet(x, 1) != iskthBitSet(x, r):
            return 0
        l -= 1
        r += 1
    return 1


def findNthPal(n):
    pal_cont = 0
    i = 0
    for i in range(maxsize):
        if isBinPal(i) > 0:
            pal_cont += 1
        if pal_cont == n:
            break
    return i


string = rl()
n = int(string, 16)
res = findNthPal(n)
wr(f'{res} -> {bin(res)[2:]}')

File: Codigos estudiantes por lenguaje/Java/Bryann Valderrama/BitWise/palindromosBinarios.java
import java.util.Scanner;

public class Main {

    static int INT_MAX = Integer.MAX_VALUE;

    static int isKthBitSet(int x, int k) {
        return ((x & (1 << (k - 1))) > 0) ? 1 : 0;
    }

    static int leftmostSetBit(int x) {
        int cont = 0;
        while (x > 0) {
            cont++;
            x = x >> 1;
        }
        return cont;
    }

    static int isBinPal(int x) {
        int l = leftmostSetBit(x);
        int r = 1;
        while (l > r) {
            if (isKthBitSet(x, 1) != isKthBitSet(x, r)) {
                return 0;
            }
            l--;
            r++;
        }
        return 1;
    }

    static int findNthPal(int n) {
        int pal_cont = 0;
        int i;
        for (i = 0; i < INT_MAX; i++) {
            if (isBinPal(i) > 0) {
                pal_cont++;
            }
            if (pal_cont == n) {
                break;
            }
        }
        return i;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        int n = Integer.parseInt(str, 16);
        int res = findNthPal(n);
        System.out.println(res + "-> " + Integer.toString(res, 2));
    }
}
